glean-core/ios/Glean/Dispatchers.swift

```diff
@@ -156,32 +160,34 @@ class Dispatchers {
     }
 
     /// Stop queuing tasks and process any tasks in the queue.
-    func flushQueuedInitialTasks() {
-        // Timeouts are easily accomplished in Swift using DispatchSemaphores and launching
-        // a block of code asynchronously.  This creates a semaphore that we can await with a
-        // timeout to prevent the queued initial tasks from hanging up or taking a long time.
-        let timeout = DispatchSemaphore(value: 0)
-        concurrentOperationsQueue.addOperation {
-            // Add all of the queued operations to the `operationQueue` which will cause them to be
-            // executed serially in the order they were collected.
-            self.serialOperationQueue.addOperations(self.preInitOperations, waitUntilFinished: true)
-            timeout.signal()
+    ///
+    /// - parameters:
+    ///     * waitUntilFinished: **USE FOR TESTING ONLY**  Boolean to force awaiting the tasks to complete
+    ///                          in order to make it easier for tests to deal with async issues.
+    func flushQueuedInitialTasks(waitUntilFinished: Bool = false) {
+        // Assert we are in testing mode if the waitUntilFinished flag is true
+        if waitUntilFinished {
+            assertInTestingMode()
         }
 
-        // Await the async task to complete up to the allowed time and log an error if it
-        // times out.
-        if timeout.wait(timeout: DispatchTime.now() + Constants.queueProcessingTimeout) == .timedOut {
-            logger.error("Timeout processing initial tasks queue")
-            // Turn off queuing to allow for normal background execution mode
-            queueInitialTasks.value = false
-            GleanMetrics.GleanError.preinitTasksTimeout.set(true)
-        } else {
-            logger.info("Initial tasks queue completed successfully")
-        }
+        // Add all of the queued operations to the `operationQueue` which will cause them to be
+        // executed serially in the order they were collected.
+        self.serialOperationQueue.addOperations(
+            self.preInitOperations,
+            waitUntilFinished: waitUntilFinished
+        )
 
         // Turn off queuing to allow for normal background execution mode
         queueInitialTasks.value = false
 
+        // This must happen after `queueInitialTasks.set(false)` is run, or it
+        // would be added to a full task queue and be silently dropped.
+        if overflowCount > 0 {
```

Comment:
If this is added, you should probably also add a test for it and a changelog entry. I'd recommend leaving this out of this PR, and take bug https://bugzilla.mozilla.org/show_bug.cgi?id=1609734 after this :)

Reply:
Okay, works for me